modules/tinymce/src/core/main/ts/newline/NewLineAction.ts

```diff
@@ -61,7 +61,7 @@ const match = (predicates, action) => {
   };
 };
 
-const getAction = (editor: Editor, evt?) => {
```

Comment:
This isn't the correct Type for this, as with ADTs we need to manually define the type. Here's what I believe would be correct:
```
export interface NewLineActionAdt {
  fold: <T> (
    br: VoidFunction,
    block: VoidFunction,
    none: VoidFunction,
  ) => T;
  match: <T> (branches: {
    br: VoidFunction;
    block: VoidFunction;
    none: VoidFunction;
  }) => T;
  log: (label: string) => void;
}

const newLineAction: {
  br: () => NewLineActionAdt;
  block: () => NewLineActionAdt;
  none: () => NewLineActionAdt;
} = Adt.generate([
  { br: [ ] },
  { block: [ ] },
  { none: [ ] }
]);
```
We can then use the `NewLineActionAdt` type here instead.

Reply:
done